### ISSUE TITLE
Raid targets for Penetrating Cold fail to be assigned when rounds of debuff are overlapping

### ISSUE THREAD

**Issue by vkosikhin** [DeadlyBossMods/DBM-Classic#774]
**Describe the bug**

DBM is supposed to assign a set of raid target icons to players affected by Penetrating Cold during the Anub'Arak fight if the corresponding option is enabled. In rare cases when the next round of the debuff starts a fraction of a second before the old round finishes, some raid targets fail to be assigned.

I propose a simple solution: remove raid targets 0.5 seconds before the active round of Penetrating Cold expires.

**Did you try having DeadlyBossMods as the only enabled addon and everything else (especially something like ElvUI) disabled?**

We disable all raid assistant roles before engaging Anub'Arak so that only one person's DBM is allowed to assign raid targets.

**Which version of DeadlyBossMods are you using?**

DBM-1.14.44-classic

**Comment by MysticalOS** [comment]
it literally has nothing to do with removing last icons. it's not going "is x icon already used, ok we can't use it then" (ironically that's actually part of the dirty "fix" for the issue, to set icons by previous set as unusable on next set, leave them up

new debuffs go out, it puts marks on them. the problem is blizzard still hasn't fixed the bug I reported yet where there is no combat log event for a refreshing buff. so even if all the icons were cleared first. the refresh wouldn't get a new icon.

it's basically going like this

bob, jerry, tom, dick, harry all had pcold
new colds go out on jane, jill, tina, terra, and harry
harry does NOT get a combat log event because harry already had it. normally spell aura refresh should fire but NOTHING does. the new icon table has 4 targets and they are jane, jill, tina, terra and they each get an icon. Harry doesn't get assigned a new icon and has an 80% chance the icon he did have got moved to someone else so you likely end up with 1 unmarked pcold.

The fix is annoying as crap and I was also worried about breaking compat with bigwigs (which might result in more icon issues in actuality) so I left code as is despite this bug. I'm gonna bring this issue up with bigwigs and see if we can come up with an answer together though but I really hate investing large amounts of time on old encounters to fix blizzard bugs. classic isn't that hard and modern addons can track buffs even if they aren't marked.

**Comment by vkosikhin** [comment]
Thank you for taking the time to answer. I realize we're not going to get a fix anytime soon. Still, I found some weak auras to be working just fine as the debuff gets reapplied despite the bug on Blizzard's part. I wonder if they use a different approach to tracking auras that is in some way incompatible with how DBM operates.

**Comment by MysticalOS** [comment]
I imagine those weak aura devs put more time into classic than I did. Probably made by players that play nothing else but classic and have all time in world for it. Classic for me will always be an afterthought after retail because I'm one person juggling too much.

none the less, i pushed a fix that should work.

**Comment by vkosikhin** [comment]
I appreciate that. I only hope that it's not going to be the 'dirty fix' that you mentioned before. :)

**Comment by MysticalOS** [comment]
I mean it's not super dirty, it's just now delayed by 0.8 seconds. it basically waits the 0.5 seconds to be sure old ones fall off, then another .3 for good measure then collects the targets that still have pcold, sorts them and sets icons

**Comment by vkosikhin** [comment]
Seems good enough to me in the way that it retains the order in which the targets get sorted for icons. Incidentally, most weak auras sort them exactly as DBM does so that everyone gets to use either tool in the same raid.

**Comment by MysticalOS** [comment]
Interesting Note. Blizzard DID fix the combat log bug I reported to them. Which is why frequency of issue was reduced, but icons were still bugging due to a race condition with DBMs icon restore code.

I have since added a disable for that to mod that should make it now behave as intended and still set icons as soon as debuffs go out, in combat log order. Once again losing raid roster sorting but I feel faster icons is better option here. Credits to funkydude from BW in helping identify bug in core that was causing the quirk and the now working combat log event for refreshing .

TL/DR. Bug should be fixed in next DBM update and BW also has bugfix for their code as well by adding refresh code. Both mods should be on same page AND working soon™

**Comment by vkosikhin** [comment]
Would it cost that much time to collect events from combat log, then sort the icons by raid roster? What I see most healers using is a combination of DBM and some WA, the latter as a primary tool for assignment and tracking. Still, people expect that if they dial in marks 1 & 2 in WA that means star and circle, not some random icons. WAs sort icons by raid roster just as DBM does currently. If they go out of sync, that's bound to cause a lot of confusion and unnecessary wipes.

**Comment by MysticalOS** [comment]
I think you're misunderstanding. it's still icons 1 2 3 4 5 in that order applied in combat log order. the sorting tech was that DBM originally also sorted marks not by order they applied in log, but by raid roster index using raid roster index value, so if a person in group 1 got debuff they got star (1) and if a person in group 5 got debuff they got icon 5

it was a nice touch so person assigned to healing star could expect their target to always be in group 1 or 2. they'll still heal star, but what group they are in isn't considered, only what order debuffs went out in combat log. this matches the bigwigs behavior. it was reason i changed it to that, cause during OG wrath and for many years dbm did use the roster index sorting not combat log order.

**Comment by vkosikhin** [comment]
You're right. I checked a recent log and see icons being assigned in the same order as the debuff goes off on different people. WA did the same thing with its numbers (my mistake thinking WAs still sort by raid roster). If it works the same way post update as you say minus the missing marks, that's all I could wish for.

**Comment by MysticalOS** [comment]
New release going out, btw I didn't realize this ticket was filed in wrong place. so I'm actually gonna close it and reference it on original issue on wrath classic repo
